src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -357,12 +363,7 @@ public CloudSdk build() {
 
       // Default SDK path
       if (sdkPath == null) {
-        Path discoveredSdkPath = new PathResolver().getCloudSdkPath();
-        if (discoveredSdkPath == null) {
-          throw new AppEngineException("Google Cloud SDK path was not provided and could not be"
-              + " found in any known install locations.");
-        }
-        sdkPath = discoveredSdkPath;
+        sdkPath = discoverSdkPath();
```

Comment:
The reasoning is scattered across a couple of issues and branches.  Rather than re-summarize here, I've opened issue #169 as I should have done earlier.  This is much more than about tests.

`ServiceLocator` is a Java thing, not an OSGi thing.  The Eclipse Buddy manifest header makes our lives slightly easier; if it's really a blocker, I'll remove it and we'll work around it.